app/webapp/src/shared/types/types.notifications.ts

```diff
@@ -0,0 +1,32 @@
+export enum ACTIVITY_EVENT_TYPE {
+  PostsParsed,
+  PostAutoposted,
+}
+
+export enum NOTIFICATION_FREQUENCY {
+  Instant,
+  Daily,
+  Weekly,
+  Monthly,
+  None,
+}
+
+export interface ActivityEvent {
+  id: string;
+  userId: string;
+  type: ACTIVITY_EVENT_TYPE;
+  timestamp: number;
+  involvedDocuments: Record<string, string[]>; // key is the collection name, value is the list of document ids
```

Comment:
my thinking here was to be able to reference multiple documents across different collections if necessary. That way we could create an activity event after performing a bulk action, for example.